server/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilder.java

```diff
@@ -65,6 +67,18 @@ public String getName() {
         return name;
     }
 
+    /**
+     * Return the field names this aggregation creates.
+     *
+     * This method is a optional helper for clients that need to know the output field names.
+     *
+     * @return The set of output field names this aggregation produces without sub-aggregation and
+     * convenience outputs ("..._as_string") or Optional.empty() if unknown or not implemented.
```

Comment:
Is it just the `..._as_string` fields we ignore?  or are there other "convenience" fields?  I'm also worried this will make it hard to test.  I would think we should be able to just run the aggregation and compare the actual output to this field list.